File: moxy-compiler/src/main/java/com/omegar/mvp/compiler/pipeline/JavaFileProcessor.java
package com.omegar.mvp.compiler.pipeline;

import com.squareup.javapoet.JavaFile;

import java.util.List;

/**
 * Date: 27-Jul-17
 * Time: 10:26
 *
 * @author Evgeny Kursakov
 */
public abstract class JavaFileProcessor<M> extends Processor<M, JavaFile> {

	@Override
	public abstract JavaFile process(M input);

}

File: moxy/compiler/src/main/java/com/omegar/mvp/compiler/pipeline/JavaFileProcessor.kt
package com.omegar.mvp.compiler.pipeline

import com.squareup.javapoet.JavaFile

/**
 * Date: 27-Jul-17
 * Time: 10:26
 *
 * @author Evgeny Kursakov
 */
abstract class JavaFileProcessor<M> : Processor<M, JavaFile>() {

    abstract override fun process(input: M): JavaFile

}